File: 0058/solution.ts
function lengthOfLastWord(s: string): number {
  let i = s.length - 1;
  while (s[i] === " ") {
    i--;
  }
  let j = i;
  while (j >= 0 && s[j] !== " ") {
    j--;
  }
  return i - j;
}

File: 0058/solution.go
package main

func lengthOfLastWord(s string) int {
	loc := len(s) - 1
	for s[loc] == ' ' {
		loc--
	}
	left := loc
	for left > 0 && s[left-1] != ' ' {
		left--
	}
	return loc - left + 1
}